api/agent.go

```diff
@@ -875,6 +875,54 @@ func (a *Agent) Monitor(loglevel string, stopCh <-chan struct{}, q *QueryOptions
 	if loglevel != "" {
 		r.params.Add("loglevel", loglevel)
 	}
+
+	_, resp, err := requireOK(a.c.doRequest(r))
+	if err != nil {
+		return nil, err
+	}
+
+	logCh := make(chan string, 64)
+	go func() {
+		defer resp.Body.Close()
+
+		scanner := bufio.NewScanner(resp.Body)
+		for {
+			select {
+			case <-stopCh:
+				close(logCh)
+				return
+			default:
+			}
+			if scanner.Scan() {
+				// An empty string signals to the caller that
+				// the scan is done, so make sure we only emit
+				// that when the scanner says it's done, not if
+				// we happen to ingest an empty line.
+				if text := scanner.Text(); text != "" {
+					logCh <- text
+				} else {
+					logCh <- " "
+				}
+			} else {
+				logCh <- ""
+			}
+		}
+	}()
+
+	return logCh, nil
+}
+
+//MonitorJSON sets the logJSON flag to true
+func (a *Agent) MonitorJSON(loglevel string, logJSON bool, stopCh <-chan struct{}, q *QueryOptions) (chan string, error) {
```

Comment:
I think you might be better suited here by having 3 functions, rather than two of them with nearly identical code.
```
func (a *Agent) Monitor(_ string, _ bool, _ <-chan struct{}, _ *QueryOptions) (chan string, error) {
  /* just call monitor(..., false)*/
}
func (a *Agent) MonitorJSON(_ string, _ bool, _ <-chan struct{}, _ *QueryOptions) (chan string, error) {
  /* just call monitor(..., true)*/
}
func (a *Agent) monitor(_ string, bool, _ <-chan struct{}, _ *QueryOptions, logJSON bool) (chan string, error) {
  /* all the code */
}
```

Reply:
Also in the version you have here now you still have the boolean param in the MonitorJSON variation which isn't necessary since the truthiness is in the function name